closure/compiler/closure_js_library.bzl

```diff
@@ -33,28 +33,49 @@ load("//closure/private:defs.bzl",
 load("//closure/compiler:closure_js_aspect.bzl",
      "closure_js_aspect")
 
-def _closure_js_library(ctx):
-  if not ctx.files.srcs and not ctx.files.externs and not ctx.attr.exports:
-    fail("Either 'srcs' or 'exports' must be specified")
-  if not ctx.files.srcs and ctx.attr.deps:
-    fail("'srcs' must be set when using 'deps', otherwise consider 'exports'")
-  if ctx.attr.language:
-    print("The closure_js_library 'language' attribute is now removed and " +
-          "is always set to " + JS_LANGUAGE_IN)
+def _maybe_declare_file(actions, file, name):
+  if file:
+    return file
+  return actions.declare_file(name)
 
-  # Create a list of the sources defined by this specific rule.
-  srcs = ctx.files.srcs
-  if ctx.files.externs:
-    print("closure_js_library 'externs' is deprecated; just use 'srcs'")
-    srcs = ctx.files.externs + srcs
+def closure_js_library_impl(
```

Comment:
There's a lot of surface area change here. I feel comfortable with this because you took the time to write more tests. Thank you. I'll do further integration testing with dependent projects afterwards, at some point.